tests/extended_temporal_memory/extensive_etm_test.py

```diff
@@ -0,0 +1,1274 @@
+    self.assertAllActiveWerePredicted()
+    self.assertAllInactiveWereUnpredicted()
+
+
+  def testE12(self):
+    """Simple learnOnOneCell test.
+
+    Train on ABCADC. Without learnOnOneCell, C should have different representations in ABC and ADC.
+    """
+    self.init({"learnOnOneCell": False})
+    self.assertFalse(self.tm.learnOnOneCell)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 10)
+    numbers[0] = 50
+    numbers[-2] = 75
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 10)
+    numbers[0] = 50
+    numbers[-2] = 75
+    proximalInputB = self.sequenceMachine.generateFromNumbers(numbers)
+
+    trainInput = proximalInputA[:-1] + proximalInputB
+
+    for _ in xrange(2):
+      self.feedTM(trainInput)
+
+    self._testTM(proximalInputA)
+    predictedActiveA = self.tm.mmGetTracePredictedActiveCells().data[-1]
+    self._testTM(proximalInputB)
+    predictedActiveB = self.tm.mmGetTracePredictedActiveCells().data[-1]
+    # check that the two representations are the same
+    self.assertNotEquals(predictedActiveA, predictedActiveB)
+
+
+  def testE13(self):
+    """Simple learnOnOneCell test.
+
+    Train on ABCADC, check that C has the same representation in ADC and ABC.
+    """
+    self.init({"learnOnOneCell": True})
+    self.assertTrue(self.tm.learnOnOneCell)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 10)
+    numbers[0] = 50
+    numbers[-2] = 75
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 10)
+    numbers[0] = 50
+    numbers[-2] = 75
+    proximalInputB = self.sequenceMachine.generateFromNumbers(numbers)
+
+    trainInput = proximalInputA[:-1] + proximalInputB
+
+    for _ in xrange(2):
+      self.feedTM(trainInput)
+
+    self._testTM(proximalInputA)
+    predictedActiveA = self.tm.mmGetTracePredictedActiveCells().data[-1]
+    self._testTM(proximalInputB)
+    predictedActiveB = self.tm.mmGetTracePredictedActiveCells().data[-1]
+    # check that the two representations are the same
+    self.assertEqual(predictedActiveA, predictedActiveB)
+
+
+  def testE14(self):
+    """learnOnOneCell with motor command should not impair behavior.
+
+    Using learnOnOneCell, does the same test as E6, using the same parameters.
+    """
+    self.init({"learnOnOneCell": True})
+    self.assertTrue(self.tm.learnOnOneCell)
+
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    proximalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+    numbers = self.sequenceMachine.generateNumbers(1, 100)
+    externalInputA = self.sequenceMachine.generateFromNumbers(numbers)
+
+    for _ in xrange(2):
+      self.feedTM(proximalInputA, activeExternalCellsSequence=externalInputA)
+
+    self._testTM(proximalInputA, activeExternalCellsSequence=externalInputA)
+
+    self.assertAllActiveWerePredicted()
```

Comment:
@subutai yes exactly. If there is just one collision between two patterns in the sequence (and, to be more specific, between the first one and another one), the sequence will have to be fed (#columns + 1) times instead of 2.